tools/c7n_slacker/c7n_slacker/slacker.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from slackclient import SlackClient
+from tenacity import retry, stop_after_attempt, wait_fixed
+
+
+class SlackBot(object):
+
+    def __init__(self, token, logger):
+        self.client = SlackClient(token)
+        self.logger = logger
+
+    @retry(stop=stop_after_attempt(2), wait=wait_fixed(2))
+    def slack_handler(self, resource_dict):
+        response = self.retrieve_user_im(resource_dict['resource_owner_value'])
```

Comment:
this should really be cached, but i think we need to reconsider if we've got a working prototype to go ahead and push this back to mailer as another delivery mechanism.